### PR TITLE
Return friendly error values.

### PR DESCRIPTION
**Pull request by defectiveAi**

### CODE REVIEW

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Services/UserIdentityOpenIdLinkService.cs [hunk 1]
```diff
@@ -35,24 +36,31 @@ public class UserIdentityOpenIdLinkService : IUserIdentityOpenIdLinkService
...
+            requestExternalUserId.Value,
+            $"External user id {requestExternalUserId} not found for open id setup.");
 
         var userIdentityInvitedOnEmail = await _userIdentityRepository.GetAsync(identityUserOpenId.Email).ConfigureAwait(false);
-
         if (userIdentityInvitedOnEmail == null)
         {
             await DeleteOpenIdUserAsync(identityUserOpenId.Id).ConfigureAwait(false);
-            throw new NotFoundValidationException($"User with email {identityUserOpenId.Email} not found with expected signInType.");
+            throw new ValidationException($"User with email {identityUserOpenId.Email} not found with expected signInType.")
```
Comment (michaelmajgaard) [inline source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Services/UserIdentityOpenIdLinkService.cs:52]: Should this not be a NotFoundException as well?

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Services/UserIdentityOpenIdLinkService.cs [hunk 2]
```diff
@@ -35,24 +36,31 @@ public class UserIdentityOpenIdLinkService : IUserIdentityOpenIdLinkService
...
+                .WithArgs(("id", identityUserOpenId.Email));
         }
 
         var userLocalIdentityByEmail = await _userRepository.GetAsync(userIdentityInvitedOnEmail.Id).ConfigureAwait(false);
-
         if (userLocalIdentityByEmail == null)
         {
             await DeleteOpenIdUserAsync(identityUserOpenId.Id).ConfigureAwait(false);
-            throw new NotFoundValidationException($"User with id {userIdentityInvitedOnEmail.Id} not found.");
+            throw new ValidationException($"User with id {userIdentityInvitedOnEmail.Id} not found.")
```
Comment (michaelmajgaard) [inline source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Services/UserIdentityOpenIdLinkService.cs:61]: Same

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Services/UserIdentityOpenIdLinkService.cs (hunk 1 above)
Comment (defectiveAi) [reply source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Services/UserIdentityOpenIdLinkService.cs:52]: No, and I am not sure it should be a ValidationException at all, actually. We are in a service, so something else har started the Link operation. Errors at this point are suspicious.

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Services/UserIdentityOpenIdLinkService.cs (hunk 2 above)
Comment (defectiveAi) [reply source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Services/UserIdentityOpenIdLinkService.cs:61]: Same

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Services/UserIdentityOpenIdLinkService.cs (hunk 1 above)
Comment (defectiveAi) [reply source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Services/UserIdentityOpenIdLinkService.cs]: These are now NotSupportedException.

**Review by michaelmajgaard — APPROVED** [review]

**Pull request merged**